c++/src/kj/compat/tls.c++

```diff
@@ -405,31 +416,108 @@ private:
 // =======================================================================================
 // Implementations of ConnectionReceiver, NetworkAddress, and Network as wrappers adding TLS.
 
-class TlsConnectionReceiver final: public kj::ConnectionReceiver {
+template <typename T>
+class InputQueue {
+  // InputQueue is an async FIFO queue that has many sources but only one sink.
+  // TODO(pr) Should I find a header home for this somewhere?
+
 public:
-  TlsConnectionReceiver(TlsContext& tls, kj::Own<kj::ConnectionReceiver> inner)
-      : tls(tls), inner(kj::mv(inner)) {}
+  void pushRaw(T&& v) {
+    // Push an existing value onto the queue.
 
-  Promise<Own<AsyncIoStream>> accept() override {
-    return inner->accept().then([this](kj::Own<AsyncIoStream> stream) {
-      return kj::evalNow([&] { return tls.wrapServer(kj::mv(stream)); })
-          .catch_([this](kj::Exception&& e) {
-        KJ_LOG(ERROR, "error accepting tls connection", e);
-        return accept();
-      });
+    if (auto localFulfiller = kj::mv(fulfiller)) {
+      // Fast path: We have a consumer waiting.
+      localFulfiller->fulfill(kj::fwd<T>(v));
+      return;
+    }
+
+    // Slow path: We don't have a consumer, add it to the queue.
+    auto node = kj::heap<Node>(kj::fwd<T>(v));
+    if (head) {
```

Comment:
Hmmmmm, perhaps we should prevent boolean casts on Own then? Maybe<Own<T>> will serve my purposes but it's definitely not as easy to use as Own<T>. I'm doing a fair bit of `(*myExtractedMaybeValue)->`.

Reply:
> Hmmmmm, perhaps we should prevent boolean casts on Own then?

Yeah, we probably should.

> Maybe<Own> will serve my purposes but it's definitely not as easy to use as Own. I'm doing a fair bit of `(*myExtractedMaybeValue)->`.

FWIW I usually do `myExtractedMaybeValue->get()->`, which is a lot easier to type.

I wish I could find a way to formulate the `KJ_IF_MAYBE(foo, bar)` macro such that `foo` ended up as a reference rather than a pointer.

Reply:
That is a lot easier to type/parse. Thanks!

I'd be interested in discussing different ways of phrasing KJ_IF_MAYBE equivalents. I doubt I've thought of every motivation you have for it, but I suspect we can satisfy at least some of them with an unnameable type that does ->/* forwarding.